homeassistant/components/stookwijzer/sensor.py

```diff
@@ -23,44 +65,40 @@ async def async_setup_entry(
     async_add_entities: AddEntitiesCallback,
 ) -> None:
     """Set up Stookwijzer sensor from a config entry."""
-    client = hass.data[DOMAIN][entry.entry_id]
-    async_add_entities([StookwijzerSensor(client, entry)], update_before_add=True)
+
+    async_add_entities(
+        StookwijzerSensor(description, entry) for description in STOOKWIJZER_SENSORS
+    )
 
 
-class StookwijzerSensor(SensorEntity):
-    """Defines a Stookwijzer binary sensor."""
+class StookwijzerSensor(CoordinatorEntity, SensorEntity):
+    """Defines a Stookwijzer sensor."""
 
-    _attr_attribution = "Data provided by stookwijzer.nu"
-    _attr_device_class = SensorDeviceClass.ENUM
+    entity_description: StookwijzerSensorDescription
+    _attr_attribution = "Data provided by atlasleefomgeving.nl"
     _attr_has_entity_name = True
-    _attr_name = None
-    _attr_translation_key = "stookwijzer"
-
-    def __init__(self, client: Stookwijzer, entry: ConfigEntry) -> None:
-        """Initialize a Stookwijzer device."""
-        self._client = client
-        self._attr_options = [cls.value for cls in StookwijzerState]
-        self._attr_unique_id = entry.entry_id
+
+    def __init__(
+        self,
+        description: StookwijzerSensorDescription,
+        entry: ConfigEntry,
+    ) -> None:
+        """Initialize the Sensor Entity."""
+        self.entity_description = description
+        self._coordinator = entry.runtime_data
+
+        super().__init__(self._coordinator)
+
+        self._attr_unique_id = f"{entry.entry_id}{DOMAIN}{description.key}"
```

Comment:
1. Why don't we use the latitude and longitude?
2. Why do we include the domain?

Reply:
> Why don't we use the latitude and longitude?

Because that is not stable, we have been removing it in other places.

> Why do we include the domain?

Already removed in my local version, makes no sense indeed.
There is a bigger problem, which is changing of unique IDs that aren't migrated.

Reply:
> Because that is not stable,

How? That's unique for the config entry right?